helix-term/src/application.rs

```diff
@@ -565,6 +569,18 @@ impl Application {
             self.editor.refresh_language_servers(id);
         }
 
+        if self.editor.config().diff_source == DiffSource::File {
```

Comment:
Yeah this should not be handled by reading the file again from disk (on the IO thread too!). Instead you should sent the current rope directly to the diff worker (altough this is pretty hacky to begin with)

Reply:
does this mean that we should always read an extra copy of the file into memory for every file we open? reading from disk means that we can avoid duplicating the file contents in memory until the user actually switches to the file-based diff, but if we only read the diff source when a file is opened, there wouldn't be any way to switch to the file-based diff source after changes have already been made to the buffer.

Reply:
I don't think k I understand what you mean here.

Ofcourse this condition should still only trigger if the config option is set appropriaietly (the check is already there) but you can simply send the  document rope to the diff worker then instead of doing IO here.

Cloneing a rope does not cause a copy of the file to be created in memory (ropes are refernebced counted and copy on write)